framework/doc/packages_config.yml

```diff
@@ -10,8 +10,8 @@ maximum_python: 3.11
 mpich: 4.0.2
 petsc_alt: 3.11.4
 petsc: 3.20.3
-moose_dev: 2024.04.23
-moose_tools: 2024.03.07
+moose_dev: 2024.04.25
+moose_tools: 2024.04.25
 apptainer_rockylinux: 8.8.20230518
 apptainer_mpich: 3.4.3
```

Comment:
Confirmed...
```bash
$ > moosetainer shell moose-dev

[moose-dev][/opt/mpich/bin]$ ./mpichversion 
MPICH Version:    	3.4.3
```
Changing this will have radical changes in our apptainer containers. Very much equivalent to what this PR is doing in our Conda packages. Unless we have a real good reason for bumping it, I suggest leaving well enough alone...

Reply:
Yeah, we can tackle that in a separate PR down the road.